src/app/Modules/auth/pages/login/login.component.spec.ts

```diff
@@ -30,4 +29,31 @@ describe('LoginComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should run #ngOnInit()', () => {
+    component.form = component.formBuilder.group({
+      email: component.email,
+      password: component.password,
+    });
+    spyOn(component.formBuilder, 'group');
+    component.ngOnInit();
+    expect(component.formBuilder.group).toHaveBeenCalled();
+  });
+
+  it('should run #getErrorMessage()', () => {
+    component.email = component.email;
+    spyOn(component.email, 'hasError');
+    component.password = component.password;
+    spyOn(component.password, 'hasError');
+    component.getErrorMessage();
+    expect(component.email.hasError).toHaveBeenCalled();
```

Comment:
Aside checking that the method is called you might want to check the return value of `getErrorMessage` to make sure it is returning the expected one.

It's not entirely clear to me what `component.email = component.email` is intended to achieve, you're assigning a variable the value it already has.

I believe a test like:
```
    spyOn(component.email, "hasError").and.callThrough();
    spyOn(component.password, "hasError").and.callThrough();
    
    component.email.setValue("foobar");
    component.password.setValue("testpassword");
    
    const res = component.getErrorMessage();
    
    expect(component.email.hasError).toHaveBeenCalledWith('required')
    expect(component.email.hasError).toHaveBeenCalledWith('email')
    expect(component.password.hasError).toHaveBeenCalledWith('required')
    expect(res).toEqual("Not a valid email");
```
will do a better job at validating that the code is working the way it's intended to.

With the current implementation I could the method under test to be:
```
// if (this.email.hasError('required') || this.password.hasError('required')) {
    //   return 'You must enter a value';
    // }
    //
    // return this.email.hasError('email') ? 'Not a valid email' : '';
    this.email.hasError(undefined)
    this.password.hasError(undefined)
    return 'The world is an amazing place'
```
and the test still pass